### PR TITLE
Respect `--configuration` option when analyzing via `pod lib lint --analyze`

### PR DESCRIPTION
**Pull request by jmagman**
#9760 introduced the `--configuration` flag and used it for `xcodebuild build`.  Also adopt for `xcodebuild analyze` for `pod lib lint --analyze` and `pod spec lint --analyze`

Fixes https://github.com/CocoaPods/CocoaPods/issues/10476

### CODE REVIEW

**Comment by jmagman** [comment]
```
spec/unit/validator_spec.rb:5:3: C: Block has too many lines. [1533/1476]
  describe Validator do ...
  ^^^^^^^^^^^^^^^^^^^^^
```
This PR pushed the `validator_spec` file over the "too many lines" lint.  How would you like me to split it up?

**Comment by igor-makarov** [comment]
Hi @jmagman! This looks exactly like https://github.com/CocoaPods/CocoaPods/pull/10238#issuecomment-787494298 - the decision is to disable this rule for specs.

**Review by paulb777 — APPROVED** [review]
LGTM - thanks!

**Comment by igor-makarov** [comment]
@segiddins  https://github.com/CocoaPods/CocoaPods/pull/10483

**Comment by igor-makarov** [comment]
@jmagman could you please rebase? This will unblock the linter error.

**Comment by jmagman** [comment]
> @jmagman could you please rebase? This will unblock the linter error.

@igor-makarov Done, thanks for #10483!

**Comment by jmagman** [comment]
@paulb777 is this waiting on anything else to be merged? No rush, just want to make sure it's not blocked on me.

**Review by segiddins — APPROVED** [review]

**Pull request merged**
